pkg/apiserver/apiserver.go

```diff
@@ -137,23 +137,33 @@ func (g *APIGroupVersion) InstallREST(container *restful.Container) error {
 // this method will return an error.
 func (g *APIGroupVersion) UpdateREST(container *restful.Container) error {
 	installer := g.newInstaller()
-	var ws *restful.WebService = nil
-
-	for i, s := range container.RegisteredWebServices() {
-		if s.RootPath() == installer.prefix {
-			ws = container.RegisteredWebServices()[i]
-			break
-		}
-	}
-
+	ws := g.GetREST(container)
 	if ws == nil {
 		return apierrors.NewInternalError(fmt.Errorf("unable to find an existing webservice for prefix %s", installer.prefix))
 	}
 	apiResources, registrationErrors := installer.Install(ws)
+	// TODO: This panics when the route is the last one in the webservice.
+	// Best effort due to this panic: https://github.com/emicklei/go-restful/issues/276
+	// Remove the supported resources route and add it back to reflect new resource adds and deletes
+	ws.RemoveRoute(ws.RootPath()+"/", "GET")
 	AddSupportedResourcesWebService(g.Serializer, ws, g.GroupVersion, apiResources)
 	return utilerrors.NewAggregate(registrationErrors)
 }
 
+// GetREST returns the REST handlers for this APIGroupVersion from an existing web service
+// in the restful Container.  It will use the prefix (root/version) to find the existing
+// web service.  If a web service does not exist within the container to support the prefix
+// this method will return nil.
+func (g *APIGroupVersion) GetREST(container *restful.Container) *restful.WebService {
```

Comment:
I was just trying to keep it in sync with the InstallRest, RemoveRest and UpdateRest functions